Cabal/src/Distribution/Simple/Program/GHC.hs

```diff
@@ -618,6 +622,76 @@ runGHC
 runGHC verbosity ghcProg comp platform opts = do
   runProgramInvocation verbosity (ghcInvocation ghcProg comp platform opts)
 
+runGHCWithResponseFile
+  :: FilePath
+  -> FilePath
+  -> Maybe TextEncoding
+  -> Verbosity
+  -> ConfiguredProgram
+  -> Compiler
+  -> Platform
+  -> GhcOptions
+  -> IO ()
+runGHCWithResponseFile workDir fileNameTemplate encoding verbosity ghcProg comp platform opts = do
+  let invocation = ghcInvocation ghcProg comp platform opts
+
+  -- Don't use response files if the first argument is `--interactive`, for
+  -- two related reasons.
+  --
+  -- `hie-bios` relies on a hack to intercept the command-line that `Cabal`
+  -- supplies to `ghc`.  Specifically, `hie-bios` creates a script around
+  -- `ghc` that detects if the first option is `--interactive` and if so then
+  -- instead of running `ghc` it prints the command-line that `ghc` was given
+  -- instead of running the command:
+  --
+  -- https://github.com/haskell/hie-bios/blob/ce863dba7b57ded20160b4f11a487e4ff8372c08/wrappers/cabal#L7
+  --
+  -- … so we can't store that flag in the response file, otherwise that will
+  -- break.  However, even if we were to add a special-case to keep that flag
+  -- out of the response file things would still break because `hie-bios`
+  -- stores the arguments to `ghc` that the wrapper script outputs and reuses
+  -- them later.  That breaks if you use a response file because it will
+  -- store an argument like `@…/ghc36000-0.rsp` which is a temporary path
+  -- that no longer exists after the wrapper script completes.
+  --
+  -- The work-around here is that we don't use a response file at all if the
+  -- first argument (and only the first argument) to `ghc` is
+  -- `--interactive`.  This ensures that `hie-bios` and all downstream
+  -- utilities (e.g. `haskell-language-server`) continue working.
+  case progInvokeArgs invocation of
+    "--interactive" : _ ->
+      runProgramInvocation verbosity invocation
+    args -> do
+      let (responseFileArgs, otherArgs) = splitArgs False args
+
+      withResponseFile
+        verbosity
+        defaultTempFileOptions
```

Comment:
Can you please add an option to keep this temporary file, in fact, is it a bug that `cabal repl` supports `--keep-temp-files` but that option is not honoured here?